packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -308,6 +309,18 @@ export class LangiumGrammarValidator {
         }
     }
 
+    checkFragmentsInTypes(grammar: ast.Grammar, accept: ValidationAcceptor): void {
+        grammar.types.map(e => {
+            e.typeAlternatives.map(e => {
+                if (ast.isParserRule(e.refType?.ref)) {
+                    if ((e.refType?.ref as ast.ParserRule).fragment) {
+                        accept('error', 'Cannot use rule fragments in as cross references in types.', { node: e, property: 'refType'});
```

Comment:
There is a typo in the error message 'Cannot use rule fragments in as cross references in types.'. Also, I'd suggest to replace it with 'Cannot use rule fragments in types'.